app/controllers/subscriptions_controller.rb

```diff
@@ -2,31 +2,36 @@ class SubscriptionsController < ApplicationController
   def create
     return render json: { id: 0 }, status: :created if smoke_test_address?
 
-    existing_subscription = nil
-    subscription = nil
+    subscription = send_email = status = nil
 
     Subscription.transaction do
+      deactivated_subscriber = subscriber.deactivated?
+      subscriber.activate! if deactivated_subscriber
+
       existing_subscription = Subscription.active.lock.find_by(
         subscriber: subscriber,
         subscriber_list: subscriber_list,
       )
 
-      existing_subscription.end(reason: :frequency_changed) if existing_subscription
+      create_new_subscription = existing_subscription.nil? ||
+        (frequency.to_s != existing_subscription.frequency.to_s)
 
-      subscriber.activate! if subscriber.deactivated?
+      if create_new_subscription
+        existing_subscription&.end(reason: :frequency_changed)
+        new_subscription = Subscription.create!(
+          subscriber: subscriber,
+          subscriber_list: subscriber_list,
+          frequency: frequency,
+          signon_user_uid: current_user.uid,
+          source: existing_subscription ? :frequency_changed : :user_signed_up,
+          )
+      end
 
-      subscription = Subscription.create!(
-        subscriber: subscriber,
-        subscriber_list: subscriber_list,
-        frequency: frequency,
-        signon_user_uid: current_user.uid,
-        source: existing_subscription ? :frequency_changed : :user_signed_up,
-      )
+      subscription = new_subscription || existing_subscription
+      send_email = create_new_subscription || deactivated_subscriber
+      status = existing_subscription.nil? ? :created : :ok
```

Comment:
Should this be `created` if `create_new_subscription`?

Reply:
That was not the behaviour of the previous code. I suppose it depends on how you think about it - if you end the previous subscription and create a new one with a different frequency you could see it as modifying the the existing subscription